libbeat/processors/convert/convert.go

```diff
@@ -403,3 +403,13 @@ func cloneValue(value interface{}) interface{} {
 		return value
 	}
 }
+
+// Helper to interpret a string as either base-10 or base-16.
+func strToInt(v string, bitSize int) (int64, error) {
+	base := 10
+	if strings.IndexAny(v, "xX") != -1 {
```

Comment:
I'm wondering if we should check to see if prefix is "0x"||"0X"? It seems odd that we would assume 0x if we found an x anywhere in the string

Reply:
Checking for a 0x prefix will require to also account for an optional -+ sign first, so we will end up duplicating most of the parsing that strconv.ParseInt does.

By just checking for an x and passing it to ParseInt, we are already sure that it's either an hex number or an not a number at all, so ParseInt will take care of that for us.

Reply:
I think a more strict check would be clearer.

Reply:
@asr That makes sense then. How about just clearing up that comment below to also include `0X` (I was reading that in a very strict sense).

Reply:
I guess you meant @adriansr.

Reply:
@asr I'm sorry. And sorry for all the times I'll probably accidentally do that in the future. 🤦‍♂ `slack handle != github handle`

Reply:
in retrospect it was silly to use strings.IndexAny on the whole string for something so simple, which has a lot of overhead. I've ended up checking for the prefix as you said 😅